v2/internal/testcommon/test_context.go

```diff
@@ -48,7 +43,7 @@ type PerTestContext struct {
 	TestContext
 	T                     *testing.T
 	logger                logr.Logger
-	AzureClientRecorder   *recorder.Recorder
+	AzureClientRecorder   testRecorder
```

Comment:
Exported field whose type is unexported seenms wrong here?

Reply:
Will fix.

Reply:
Fixed by introduction of subpackages.